[GStreamer] Add transfer annotations to functions that return element pointer

ntrrgc opened the pull request:
#### d259cb2a0dedc8005483df441736c7e92f68c056
<pre>
[GStreamer] Add transfer annotations to functions that return element pointer
<a href="https://bugs.webkit.org/show_bug.cgi?id=291898">https://bugs.webkit.org/show_bug.cgi?id=291898</a>

Reviewed by Philippe Normand.

Instances of GstElement must be floating on construction.

This is an API requirement from GstElementFactory (see
gst_element_factory_create_with_properties() for details), and it is
automatically fulfilled by the constructor of GInitiallyUnowned, which
is a base class of GstElement.

However, this may not be immediately obvious to readers, so this patch
adds transfer annotations in comments to a few functions in
GStreamerCommon that return raw pointers.

This patch adds comments and assertions and introduces no new behavior.

Potentially converting those functions to use GRefPtr instead of raw
pointers is outside of the scope of this patch.

* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::createAutoAudioSink):
(WebCore::createPlatformAudioSink):
(WebCore::gstBufferNewWrappedFast):
(WebCore::makeGStreamerElement):
* Source/WebCore/platform/graphics/gstreamer/WebKitAudioSinkGStreamer.cpp:
(webkitAudioSinkNew):

Canonical link: <a href="https://commits.webkit.org/294188@main">https://commits.webkit.org/294188@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b1344b995d6f8c396d2e121ab54fa1a6a9fad456

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/100424 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/20076 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/10375 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/105561 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51012 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/102465 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/20382 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/28550 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/76466 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/33518 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103431 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/15614 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/90713 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56822 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/15430 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/8715 "Passed tests") | [❌ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/50386 "Failed to compile WebKit") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/85346 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/8795 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/107915 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27542 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20209 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/85419 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/27905 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86913 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84956 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21761 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/29640 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/7372 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/21493 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27477 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/32720 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27288 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30606 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/28846 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b1344b995d6f8c396d2e121ab54fa1a6a9fad456. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/44372

eocanha commented:
We need more of this! Informal r+

philn commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
```diff
@@ -1007,10 +1007,11 @@ IGNORE_WARNINGS_END
 }
 
-GstElement* createPlatformAudioSink(const String& role)
+GstElement* /* (transfer floating) */ createPlatformAudioSink(const String& role)
```
No `ASSERT(g_object_is_floating(audioSink));` here?

ntrrgc replied:
It is already in the two constructors used inside. If you think I should add it here anyway, I can do that too.

philn replied:
Ah OK, I think it's not needed then :) r+

ntrrgc committed:
[GStreamer] Add transfer annotations to functions that return element pointer
https://bugs.webkit.org/show_bug.cgi?id=291898

Reviewed by Philippe Normand.

Instances of GstElement must be floating on construction.

This is an API requirement from GstElementFactory (see
gst_element_factory_create_with_properties() for details), and it is
automatically fulfilled by the constructor of GInitiallyUnowned, which
is a base class of GstElement.

However, this may not be immediately obvious to readers, so this patch
adds transfer annotations in comments to a few functions in
GStreamerCommon that return raw pointers.

This patch adds comments and assertions and introduces no new behavior.

Potentially converting those functions to use GRefPtr instead of raw
pointers is outside of the scope of this patch.

* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::createAutoAudioSink):
(WebCore::createPlatformAudioSink):
(WebCore::gstBufferNewWrappedFast):
(WebCore::makeGStreamerElement):
* Source/WebCore/platform/graphics/gstreamer/WebKitAudioSinkGStreamer.cpp:
(webkitAudioSinkNew):

Canonical link: https://commits.webkit.org/294188@main

webkit-commit-queue commented:
Committed 294188@main (d259cb2a0ded): <https://commits.webkit.org/294188@main>

Reviewed commits have been landed. Closing PR #44372 and removing active labels.
